lua/lspconfig/configs/gh_actions_ls.lua

```diff
@@ -9,9 +9,19 @@ return {
     -- files. (A nil root_dir and no single_file_support results in the LSP not
     -- attaching.) For details, see #3558
     root_dir = function(filename)
-      return filename:find('/%.(github|forgejo|gitea)/workflows/.+%.ya?ml')
-          and util.root_pattern('.github', '.forgejo', '.gitea')(filename)
-        or nil
+      local patterns = {
+        { pattern = '/%.github/workflows/.+%.ya?ml', root = '.github' },
+        { pattern = '/%.forgejo/workflows/.+%.ya?ml', root = '.forgejo' },
+        { pattern = '/%.gitea/workflows/.+%.ya?ml', root = '.gitea' },
+      }
+
+      for _, entry in ipairs(patterns) do
+        if filename:find(entry.pattern) then
+          return util.root_pattern(entry.root)(filename)
```

Comment:
using vim.fs.find on there please see `:help vim.fs.find`

Reply:
This is how I changed in my PR: https://github.com/dundargoc/nvim-lspconfig/blob/refactor/root_pattern/lua/lspconfig/configs/gh_actions_ls.lua#L1-L13. Just copy that and adjust as necessary.

Reply:
Wait, what is actually happening here though? What's with all the patterns and stuff? Does the following not solve the problem?

```lua
root_dir = function(fname)
  return vim.fs.dirname(vim.fs.find({ '.github', '.forgejo', '.gitea' }, { path = fname, upward = true })[1])
end,
```

Reply:
By the way, there were complications with the root_pattern PR so you can keep using it for now if you still want to. I'll take a look at codespell warning.

Reply:
Will merge when green: https://github.com/neovim/nvim-lspconfig/pull/3577

Reply:
> What do you mean by "complications" ?

Unrelated to your PR: https://github.com/neovim/nvim-lspconfig/pull/3569#issuecomment-2605033349

> Do I need to change anything ?

No, you don't need to change anything. This looks fine.